manticore/core/cpu/abstractcpu.py

```diff
@@ -570,27 +676,40 @@ def fallback_to_emulate(*operands):
             logger.info("Unimplemented instruction: 0x%016x:\t%s\t%s\t%s",
                     instruction.address, text_bytes, instruction.mnemonic,
                     instruction.op_str)
+            #broadcast event
+            self.will_emulate_instruction(instruction)
+
             self.emulate(instruction)
 
+            #broadcast event
+            self.did_emulate_instruction(instruction)
+
         implementation = getattr(self, name, fallback_to_emulate)
 
         if logger.level == logging.DEBUG :
             logger.debug(self.render_instruction())
             for l in self.render_registers():
                 register_logger.debug(l)
-
+        
         implementation(*instruction.operands)
         self._icount+=1
 
+        #broadcast event
```

Comment:
i'd remove all the 'broadcast event' comments, as the naming convention is pretty clear and they add significant extra lines, imho. (also okay with leaving them in)